src/map.cpp

```diff
@@ -1709,7 +1709,7 @@ bool map::furn_set( const tripoint &p, const furn_id &new_furniture, const bool
     invalidate_max_populated_zlev( p.z );
 
     set_memory_seen_cache_dirty( p );
-    if( player_character.sees( p ) ) {
+    if( inbounds( player_character.pos_bub() ) && player_character.sees( p ) ) {
```

Comment:
I don't think this is correct. `pos_bub()` returns position relative to the main map, but the map running this function at mapgen is just a temporary tinymap. So the coordinate origins don't necessarily line up.

`player_character.sees( p )` also only makes sense if this map is the main map. In both cases the tripoint needs to be converted to absolute from this map, then to local coordinates for `get_map()`

Reply:
I agree this isn't technically "correct", but I don't see an obvious way to call sees() with the bounds check but without making every non-mapgen caller pay the perf cost for it; Character::sees() calls into get_map().pl_sees() that reads out of bounds at line 799 during mapgen because the avatar isn't yet initialized
https://github.com/CleverRaven/Cataclysm-DDA/blob/d505f22d68e4ea1f0bb5630de74365c06f241900/src/lightmap.cpp#L792-L799

pos_bub will return relative to main map but that is what that function is checking

When ter_set/furn_set is called during mapgen the avatar position isn't in bounds of any map including the main one (it's hardcoded 20,10,-500 in Creature ctor, and at some point before world loads changes to -4768,-4634,-500)

Another option is a dynamic_cast/typeid hack, since the map/tinymap inheritance is somehow flipped around

Or just paying the price of a bounds check inside, but have it useless for most calls after game is past loading

Do you or anyone else have a better solution or opinion on one?